From 9ae3f94209c647f414347e96f7e95fa81a31a559 Mon Sep 17 00:00:00 2001
From: KangDroid <hyunwoo286@gmail.com>
Date: Fri, 8 May 2015 21:27:29 +0100
Subject: [PATCH 08/23] Fix-OpenMP-support-on-Android

---
 gcc/config/linux-android.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gcc/config/linux-android.h b/gcc/config/linux-android.h
index 7b3a1e5..8282217 100644
--- a/gcc/config/linux-android.h
+++ b/gcc/config/linux-android.h
@@ -49,7 +49,8 @@
   "%{!frtti:%{!fno-rtti: -fno-rtti}}"
 
 #define ANDROID_LIB_SPEC \
-  "%{!static: -ldl}"
+  "%{!static: -ldl} \
+  %{pthread: -lc}"
 
 #define ANDROID_STARTFILE_SPEC						\
   "%{shared: crtbegin_so%O%s;:"						\
-- 
2.3.2 (Apple Git-55)

