From b3dd6db863235bea3319a8fc30d2221da0652e66 Mon Sep 17 00:00:00 2001
From: KangDroid <hyunwoo286@gmail.com>
Date: Fri, 8 May 2015 21:31:00 +0100
Subject: [PATCH 15/23] Add-fuse-ld-mcld

---
 gcc/common.opt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/gcc/common.opt b/gcc/common.opt
index 51833c1..074d2ca 100644
--- a/gcc/common.opt
+++ b/gcc/common.opt
@@ -2405,9 +2405,13 @@ Common Driver Negative(fuse-ld=gold)
 Use the bfd linker instead of the default linker
 
 fuse-ld=gold
-Common Driver Negative(fuse-ld=bfd)
+Common Driver Negative(fuse-ld=mcld)
 Use the gold linker instead of the default linker
 
+fuse-ld=mcld
+Common Driver Negative(fuse-ld=bfd)
+Use the mcld linker instead of the default linker
+
 fuse-linker-plugin
 Common Undocumented Var(flag_use_linker_plugin)
 
-- 
2.3.2 (Apple Git-55)

