--- gcc-linaro-4.9-2014.08/gcc/config/aarch64/aarch64-linux.h.omv~	2014-08-16 19:56:45.446340606 +0200
+++ gcc-linaro-4.9-2014.08/gcc/config/aarch64/aarch64-linux.h	2014-08-16 19:56:55.631213277 +0200
@@ -22,6 +22,7 @@
 #define GCC_AARCH64_LINUX_H
 
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-aarch64%{mbig-endian:_be}%{mabi=ilp32:_ilp32}.so.1"
+#undef BIONIC_DYNAMIC_LINKER
 #define BIONIC_DYNAMIC_LINKER "/system/bin/linker64"
 
 #define CPP_SPEC "%{pthread:-D_REENTRANT}"
