--- gcc/gcc/input.c.jj	2014-01-24 16:32:34.000000000 +0100
+++ gcc/gcc/input.c	2014-01-24 16:41:42.012671452 +0100
@@ -698,7 +698,13 @@ location_get_source_line (expanded_locat
   static char *buffer;
   static ssize_t len;
 
-  fcache * c = lookup_or_add_file_to_cache_tab (xloc.file);
+  if (xloc.line == 0)
+    return NULL;
+
+  fcache *c = lookup_or_add_file_to_cache_tab (xloc.file);
+  if (c == NULL)
+    return NULL;
+
   bool read = read_line_num (c, xloc.line, &buffer, &len);
 
   if (read && line_len)
diff --git a/gcc/config/gnu-user.h b/gcc/config/gnu-user.h
index a1955a7..d131e7d 100644
--- a/gcc/config/gnu-user.h
+++ b/gcc/config/gnu-user.h
@@ -110,7 +110,9 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 #define USE_LD_AS_NEEDED 1
 #endif
 
+#ifndef TARGET_POSIX_IO
 #define TARGET_POSIX_IO
+#endif
 
 #undef TARGET_LIBC_HAS_FUNCTION
 #define TARGET_LIBC_HAS_FUNCTION gnu_libc_has_function
