--- gcc-4.9/gcc/config/aarch64/aarch64-linux.h.p22~	2014-03-06 07:32:30.072980899 +0100
+++ gcc-4.9/gcc/config/aarch64/aarch64-linux.h	2014-03-06 07:31:07.994204212 +0100
@@ -22,6 +22,7 @@
 #define GCC_AARCH64_LINUX_H
 
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-aarch64%{mbig-endian:_be}.so.1"
+#undef BIONIC_DYNAMIC_LINKER // Unset the /system/bin/linker default
 #define BIONIC_DYNAMIC_LINKER "/system/bin/linker64"
 
 #define CPP_SPEC "%{pthread:-D_REENTRANT}"
