--- gcc-4.9/gcc/config/arm/linux-eabi.h.p20~	2014-03-05 01:17:10.000000000 +0100
+++ gcc-4.9/gcc/config/arm/linux-eabi.h	2014-03-06 04:37:39.786126091 +0100
@@ -91,7 +91,7 @@
 #define CC1_SPEC							\
   LINUX_OR_ANDROID_CC (GNU_USER_TARGET_CC1_SPEC " " ASAN_CC1_SPEC,	\
 		       GNU_USER_TARGET_CC1_SPEC " " ASAN_CC1_SPEC " "	\
-		       ANDROID_CC1_SPEC)
+		       ANDROID_CC1_SPEC("-fpic"))
 
 #define CC1PLUS_SPEC \
   LINUX_OR_ANDROID_CC ("", ANDROID_CC1PLUS_SPEC)
--- gcc-4.9/gcc/config/i386/linux-common.h.p20~	2014-03-05 01:17:10.000000000 +0100
+++ gcc-4.9/gcc/config/i386/linux-common.h	2014-03-06 04:37:39.786126091 +0100
@@ -30,7 +30,14 @@ along with GCC; see the file COPYING3.
 #undef CC1_SPEC
 #define CC1_SPEC \
   LINUX_OR_ANDROID_CC (GNU_USER_TARGET_CC1_SPEC, \
-		       GNU_USER_TARGET_CC1_SPEC " " ANDROID_CC1_SPEC)
+                       GNU_USER_TARGET_CC1_SPEC \
+                       " -mstackrealign -msse3" \
+                       " -m32 -fno-short-enums" \
+                       " " \
+                       ANDROID_CC1_SPEC("-fPIC"))
+
+#define CC1PLUS_SPEC \
+  LINUX_OR_ANDROID_CC ("", ANDROID_CC1PLUS_SPEC)
 
 #undef	LINK_SPEC
 #define LINK_SPEC \
--- gcc-4.9/gcc/config/linux-android.h.p20~	2014-03-06 04:37:01.368833542 +0100
+++ gcc-4.9/gcc/config/linux-android.h	2014-03-06 04:37:39.786126091 +0100
@@ -40,9 +40,9 @@
 #define ANDROID_LINK_SPEC \
   "%{shared: -Bsymbolic}"
 
-#define ANDROID_CC1_SPEC						\
+#define ANDROID_CC1_SPEC(ANDROID_PIC_DEFAULT)				\
   "%{!mglibc:%{!muclibc:%{!mbionic: -mbionic}}} "			\
-  "%{!fno-pic:%{!fno-PIC:%{!fpic:%{!fPIC: -fPIC}}}}"
+  "%{!fno-pic:%{!fno-PIC:%{!fpic:%{!fPIC: " ANDROID_PIC_DEFAULT "}}}}"
 
 #define ANDROID_CC1PLUS_SPEC						\
   "%{!fexceptions:%{!fno-exceptions: -fno-exceptions}} "		\
