--- gcc-5.0/gcc/config/aarch64/aarch64-linux.h.omv~	2015-01-03 13:17:54.257592402 +0100
+++ gcc-5.0/gcc/config/aarch64/aarch64-linux.h	2015-01-03 13:18:04.830983417 +0100
@@ -22,6 +22,7 @@
 #define GCC_AARCH64_LINUX_H
 
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-aarch64%{mbig-endian:_be}%{mabi=ilp32:_ilp32}.so.1"
+#undef BIONIC_DYNAMIC_LINKER
 #define BIONIC_DYNAMIC_LINKER "/system/bin/linker64"
 
 #undef  ASAN_CC1_SPEC
