--- gcc-5.0/gcc/graphite-interchange.c.omv~	2015-01-04 01:27:45.630147199 +0100
+++ gcc-5.0/gcc/graphite-interchange.c	2015-01-04 01:28:08.640319385 +0100
@@ -32,11 +32,11 @@ along with GCC; see the file COPYING3.
 #include <isl/val.h>
 
 /* Since ISL-0.13, the extern is in val_gmp.h.  */
-#if !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
+#if 0 && !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
 extern "C" {
 #endif
 #include <isl/val_gmp.h>
-#if !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
+#if 0 && !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
 }
 #endif
 #endif
--- gcc-5.0/gcc/graphite-isl-ast-to-gimple.c.omv~	2015-01-04 01:27:09.036539971 +0100
+++ gcc-5.0/gcc/graphite-isl-ast-to-gimple.c	2015-01-04 01:27:24.336654498 +0100
@@ -27,11 +27,11 @@ along with GCC; see the file COPYING3.
 #include <isl/ast_build.h>
 
 /* Since ISL-0.13, the extern is in val_gmp.h.  */
-#if !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
+#if 0 && !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
 extern "C" {
 #endif
 #include <isl/val_gmp.h>
-#if !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
+#if 0 && !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
 }
 #endif
 #endif
--- gcc-5.0/gcc/graphite-optimize-isl.c.omv~	2015-01-04 01:28:37.220533213 +0100
+++ gcc-5.0/gcc/graphite-optimize-isl.c	2015-01-04 01:29:08.980770781 +0100
@@ -460,7 +460,7 @@ getScheduleForBandList (isl_band_list *B
 	      if (flag_loop_unroll_jam && (i != (ScheduleDimensions - depth)))
 		continue;
 
-#ifdef HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE
+#if 1 || defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE)
 	      if (isl_band_member_is_coincident (Band, i))
 #else
 	      if (isl_band_member_is_zero_distance (Band, i))
--- gcc-5.0/gcc/graphite-poly.c.omv~	2015-01-04 01:29:27.760911230 +0100
+++ gcc-5.0/gcc/graphite-poly.c	2015-01-04 01:29:38.414324228 +0100
@@ -31,11 +31,11 @@ along with GCC; see the file COPYING3.
 #include <isl/val.h>
 
 /* Since ISL-0.13, the extern is in val_gmp.h.  */
-#if !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
+#if 0 && !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
 extern "C" {
 #endif
 #include <isl/val_gmp.h>
-#if !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
+#if 0 && !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
 }
 #endif
 #endif
--- gcc-5.0/gcc/graphite-sese-to-poly.c.omv~	2015-01-04 01:29:55.821121042 +0100
+++ gcc-5.0/gcc/graphite-sese-to-poly.c	2015-01-04 01:30:06.581194740 +0100
@@ -29,11 +29,11 @@ along with GCC; see the file COPYING3.
 #include <isl/val.h>
 
 /* Since ISL-0.13, the extern is in val_gmp.h.  */
-#if !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
+#if 0 && !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
 extern "C" {
 #endif
 #include <isl/val_gmp.h>
-#if !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
+#if 0 && !defined(HAVE_ISL_SCHED_CONSTRAINTS_COMPUTE_SCHEDULE) && defined(__cplusplus)
 }
 #endif
 #endif
