--- gcc-5.0/gcc/cp/tree.c.omv~	2015-01-08 23:31:38.623082438 +0100
+++ gcc-5.0/gcc/cp/tree.c	2015-01-08 23:32:26.797107305 +0100
@@ -875,12 +875,12 @@ build_cplus_array_type (tree elt_type, t
 	{
 	  t = build_min_array_type (elt_type, index_type);
 	  set_array_type_canon (t, elt_type, index_type);
-	  if (!dependent)
-	    layout_type (t);
 
 	  TYPE_MAIN_VARIANT (t) = m;
 	  TYPE_NEXT_VARIANT (t) = TYPE_NEXT_VARIANT (m);
 	  TYPE_NEXT_VARIANT (m) = t;
+	  if (!dependent)
+	    layout_type (t);
 	}
     }
 
